client/src/hooks/useForgotPasswordForm.jsx

```diff
@@ -0,0 +1,103 @@
+import { useDisclosure, useToast } from '@chakra-ui/react';
+import { useRef, useState } from 'react';
+import useEmailValidation from 'hooks/useEmailValidation.jsx';
+import { userForgotPassword } from 'utils/api/users.js';
+import { sendEmail } from 'utils/api/system.js';
+
+const useForgotPasswordForm = () => {
+  const emailRef = useRef('');
+  const [emailIsValidated, setEmailIsValidated] = useState(true);
+  const [humanIsValidated, setHumanIsValidated] = useState(true);
+  const [buttonIsDisabled, setButtonIsDisabled] = useState(true);
+  const toast = useToast();
+  const [isLoading, setIsLoading] = useState(false);
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  const { isAnEmail } = useEmailValidation();
+
+  const forgotPasswordHandleSubmit = async (e) => {
+    e.preventDefault();
+    setIsLoading(true);
+
+    const email = emailRef.current.value;
+
+    try {
```

Comment:
I don't think you need nested try catch. You can use one catch clause and have if/else conditions and display message based on condition.

Reply:
its fixed